src/lib/Scenes/Home/index.tsx

```diff
@@ -49,12 +61,22 @@ export default class Home extends React.Component<Props, State> {
     AppState.removeEventListener("change", this._handleAppStateChange)
   }
 
+  // FIXME: Make a proper "viewDidAppear" callback / event emitter
+  // https://github.com/artsy/emission/issues/930
+  // This is called when the overall home component appears in Eigen
+  // We use it to dispatch screen events at that point
+  componentWillReceiveProps(newProps) {
+    if (newProps.isVisible) {
+      this.fireHomeScreenViewAnalytics()
+    }
+  }
+
   _handleAppStateChange = nextAppState => {
     // If we are coming back to the app from the background with a selected artist, make sure we're on the Artists tab
     if (this.props.selectedArtist && this.state.appState.match(/inactive|background/) && nextAppState === "active") {
       this.tabView.goToPage(0)
     }
-    this.setState({ appState: nextAppState })
+    this.setState({ appState: nextAppState, selectedTab: 0 }, () => this.fireHomeScreenViewAnalytics())
```

Comment:
If you define the analytics method like so:

```js
fireHomeScreenViewAnalytics = () => {
  // ...
}
```

…then it will basically do `this.fireHomeScreenViewAnalytics.bind(this)`, which means you can omit the extra closure:

```js
this.setState({ appState: nextAppState, selectedTab: 0 }, this.fireHomeScreenViewAnalytics)
```

Just a little thing to make your JS more idiomatic :)

Reply:
done